### PR TITLE
Ogre Beards and Hairstyles

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:
- Added a set of beards and hairstyles for the ogre ethnicity
- Added portrait modifier: 80% of ogres are bald and clean-shaven.

# How to test:
- Provide feedback on the chosen beard and hairstyle options added for male and female ogres by generating random ogres in the Ruler Designer.
- Provide feedback on the rate (80%) of bald and clean-shaven ogres. Should it be higher or lower?

### CODE REVIEW

**Comment by TheDrPibb1999** [comment]
Every ogre i have generated has had a beard, I think the 80% for clean shaven is a good idea, not too common in the game

**Comment by ercarp** [comment]
> 
> 
> Every ogre i have generated has had a beard, I think the 80% for clean shaven is a good idea, not too common in the game

My bad, I think I wasn't clear on this. The clean-shaven and baldness modifiers aren't taken into consideration when you generate ogres using the Ruler Designer or Debug Portrait Editor. This is something that's calculated in-game. If you load into a game and look for ogres in the game world, you should see the 80% modifier come into action.

**Review by TheGroxEmpire — COMMENTED** [review]
You're right about needing to make custom hairstyle for Ogres. The Vanilla hairstyle are definitely not workable at all. The hairs are floating and clipping through the horn most of the time:
<details>

image
image
image
</details>

I don't thing this is fixable without changing much of their current shape, but I'm happy to be proven wrong.

**Comment by zumbak04** [comment]
> You're right about needing to make custom hairstyle for Ogres. The Vanilla hairstyle are definitely not workable at all. The hairs are floating and clipping through the horn most of the time:
> 
> I don't think this is fixable without changing much of their current shape, but I'm happy to be proven wrong.

True. So I took a closer look, these hairstyles work well...
`male_hair_northern_01` AKA Very Long Wide
`male_hair_western_04` AKA Fade
`male_hair_indian_01_wc` AKA Bald Top

File: common/genes/03_genes_accessories_beards.txt [hunk 1]
```diff
@@ -303,5 +303,20 @@ accessory_genes = {
...
+				15 = male_beard_northern_03
+				15 = male_beard_northern_02
+				15 = male_beard_northern_01
+				15 = male_beard_western_07
+				5 = male_beard_mena_03
+			}
+			female = {
+				1 = empty
+			}
+		}
```
Comment (zumbak04) [inline common/genes/03_genes_accessories_beards.txt:320]: These are orcish beards except `male_beard_steppe_02`. Maybe we don't need this gene and the ogres will use the orc beards?
What to do with `male_beard_steppe_02` is up to you. You may include it in the orcish beards.

Comment (ercarp) [reply common/genes/03_genes_accessories_beards.txt:320]: > 
> 
> These are orcish beards except `male_beard_steppe_02`. Maybe we don't need this gene and the ogres will use the orc beards?
> What to do with `male_beard_steppe_02` is up to you. You may include it in the orcish beards.

I added it for a reason. Ogres are canonically supposed to be able to grow hair on their upper lip as well, so I wanted to have one beard like that. As you can see, I made it much rarer to occur than the other beards. Not to mention ogres having a beard, in general, is quite rare.

But I suppose we could just use the orc beards and add `male_beard_steppe_02` into the mix as a rare alternative.

Comment (ercarp) [reply common/genes/03_genes_accessories_beards.txt]: Addressed this in f7b06f9492a29674102d6c15e20dc8c9f1a4de7b.

File: common/ethnicities/wc_ethnicities_ogre.txt [hunk 2]
```diff
@@ -383,10 +383,10 @@ ogre_ethnicity = {
```
Comment (zumbak04) [inline common/ethnicities/wc_ethnicities_ogre.txt:383]: What about limiting hairstyles to the ones that don't clip or fly?
https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/188#pullrequestreview-595418862

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/ethnicities/wc_ethnicities_ogre.txt (hunk 2 above)
Comment (ercarp) [reply common/ethnicities/wc_ethnicities_ogre.txt:383]: I personally don't think clipping is such a big deal (played too much WoW so I'm used to it) but can do. 👌

Comment (ercarp) [reply common/ethnicities/wc_ethnicities_ogre.txt:383]: Taken care of in 3cb6fea594db534be0e8c1ebd5ecfd4ffadc4d5f!

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Pull request merged**